File: Problem2.js
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None


// Your code here along with comments explaining your approach
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
 var merge = function(nums1, m, nums2, n) {
    while (n) {
      if (nums1[m - 1] > nums2[n - 1]) {
        nums1[m + n - 1] = nums1[--m];  
      } else {
        nums1[m + n - 1] = nums2[--n];   
      }
    }
    return nums1;
  };

File: problem2.go
// Time Complexity : O(m*n)
// Space Complexity : O(1) 
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No
func merge(nums1 []int, m int, nums2 []int, n int)  {
    if len(nums1)==0 {
        return 
    }
    
    p1 := m-1
    p2 := n-1
    p3 := m+n-1
    
    for p1 >=0 && p2>=0{
        if nums1[p1]>=nums2[p2]{
            nums1[p3] = nums1[p1]
            p1-=1
        }else{
            nums1[p3] = nums2[p2]
            p2-=1
        }
        
        p3-=1
    }
    
    
    for p2 >= 0{
        nums1[p3] = nums2[p2]
        p2-=1
        p3-=1
    }
    
    
}
